### ISSUE TITLE
Regular InstanceNotFoundException: org.apache.cassandra.internal:type=AntiEntropySessions

### ISSUE THREAD

**Issue by vrischmann** [thelastpickle/cassandra-reaper#155]
Hello,

I'm running the latest commit of cassandra reaper and after launching an incremental repair on one keyspace I regularly see exceptions like this in the logs:

    DEBUG  [2017-08-09 12:25:00,003] [my_cluster:d6b18350-7cfc-11e7-ab40-0d91502a8808:d6b37f20-7cfc-11e7-ab40-0d91502a8808] c.s.r.c.JmxProxy - No repair has run yet on the node. Ignoring exception.
    javax.management.InstanceNotFoundException: org.apache.cassandra.internal:type=AntiEntropySessions
            at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getMBean(DefaultMBeanServerInterceptor.java:1095)
            at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getAttribute(DefaultMBeanServerInterceptor.java:643)
            at com.sun.jmx.mbeanserver.JmxMBeanServer.getAttribute(JmxMBeanServer.java:678)
            at javax.management.remote.rmi.RMIConnectionImpl.doOperation(RMIConnectionImpl.java:1445)
            at javax.management.remote.rmi.RMIConnectionImpl.access$300(RMIConnectionImpl.java:76)
            at javax.management.remote.rmi.RMIConnectionImpl$PrivilegedOperation.run(RMIConnectionImpl.java:1309)
            at javax.management.remote.rmi.RMIConnectionImpl.doPrivilegedOperation(RMIConnectionImpl.java:1401)
            at javax.management.remote.rmi.RMIConnectionImpl.getAttribute(RMIConnectionImpl.java:639)
            at sun.reflect.GeneratedMethodAccessor61.invoke(Unknown Source)
            at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
            at java.lang.reflect.Method.invoke(Method.java:498)
            at sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:357)
            at sun.rmi.transport.Transport$1.run(Transport.java:200)
            at sun.rmi.transport.Transport$1.run(Transport.java:197)
            at java.security.AccessController.doPrivileged(Native Method)
            at sun.rmi.transport.Transport.serviceCall(Transport.java:196)
            at sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:568)
            at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:826)
            at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCPTransport.java:683)
            at java.security.AccessController.doPrivileged(Native Method)
            at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:682)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
            at java.lang.Thread.run(Thread.java:748)
            at sun.rmi.transport.StreamRemoteCall.exceptionReceivedFromServer(StreamRemoteCall.java:276)
            at sun.rmi.transport.StreamRemoteCall.executeCall(StreamRemoteCall.java:253)
            at sun.rmi.server.UnicastRef.invoke(UnicastRef.java:162)
            at com.sun.jmx.remote.internal.PRef.invoke(Unknown Source)
            at javax.management.remote.rmi.RMIConnectionImpl_Stub.getAttribute(Unknown Source)
            at javax.management.remote.rmi.RMIConnector$RemoteMBeanServerConnection.getAttribute(RMIConnector.java:903)
            at com.spotify.reaper.cassandra.JmxProxy.isRepairRunningPre22(JmxProxy.java:355)
            at com.spotify.reaper.cassandra.JmxProxy.isRepairRunning(JmxProxy.java:344)
            at com.spotify.reaper.service.SegmentRunner.canRepair(SegmentRunner.java:297)
            at com.spotify.reaper.service.SegmentRunner.runRepair(SegmentRunner.java:184)
            at com.spotify.reaper.service.SegmentRunner.run(SegmentRunner.java:101)
            at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
            at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:108)
            at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:41)
            at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:77)
            at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
            at java.util.concurrent.FutureTask.run(FutureTask.java:266)
            at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
            at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
            at java.lang.Thread.run(Thread.java:748)

It's reported in DEBUG and my repairs still finish so I'm assuming they're harmless ?

Still, I'm wondering if there's something wrong with my setup here ?

I'm running a 4 node 2.2.10 cluster, and as far as I can remember I never had these exceptions when the cluster was in 2.1.x, but I may be mistaken.

Thank you.

**Comment by adejanovski** [comment]
It's actually normal. The AntiEntropySession mbean only appear if repair have been running at least once on Cassandra nodes before 2.2.  After 2.2 if I'm correct it changed but we still test all possibilities for all clusters whatever the version. 

TL;DR : The exception is harmless and won't affect repair progress.

**Comment by vrischmann** [comment]
Ok thanks @adejanovski good to know
